kiwix-build.py

```diff
@@ -399,7 +400,14 @@ def download(self, what, where=None):
             if what.sha256 == get_sha256(file_path):
                 raise SkipCommand()
             os.remove(file_path)
-        urllib.request.urlretrieve(file_url, file_path)
+        if options.no_cert_check == True:
+            context = ssl.create_default_context()
+            context.check_hostname = False
+            context.verify_mode = ssl.CERT_NONE
+            with urllib.request.urlopen(file_url, context=context) as resource, open(file_path, 'wb') as file:
+                file.write(resource.read())
```

Comment:
Here we will read all the content at once and write it once in the file.
On big archive, this meanr that we will use a lot of memory just to handle a temporary buffer.

It should be better to copy the content by small chunks, ones after the others.

For most of dependencies, archives are pretty small and it is not a problem but android tools (ndk, sdk) are pretty big (more than 600Mo)

(To be honest, I don't know if `urlretrieve` handles this problem correctly, I cannot find any clues in the documentation. But now, if we handle inner mechanics ourselves, we have to handle this point)